src/gg2rdf.ts

```diff
@@ -0,0 +1,1208 @@
+    const errors: string[] = [];
+    const taxon: Element | undefined = document.querySelector(
+      'document treatment subSubSection[type="nomenclature"] taxonomicName',
+    );
+    if (!taxon) {
+      errors.push("the treatment is lacking the taxon");
+    } else {
+      const rank = taxon.getAttribute("rank");
+      if (!rank) {
+        errors.push("the treatment taxon is lacking its rank attribute");
+      }
+      const sigEpithet = normalizeSpace(taxon.getAttribute(rank)); // get the attribute with the rank as the name
+      if (sigEpithet.match(/[^a-zA-Z\-]/)) {
+        errors.push(`sigEpithet '${sigEpithet}' contains invalid characters`);
+      }
+      if (
+        (rank === "subSpecies" || rank === "variety") &&
+        normalizeSpace(taxon.getAttribute("species")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `species '${
+            normalizeSpace(taxon.getAttribute("species"))
+          }' contains invalid characters`,
+        );
+      }
+      if (
+        (rank === "subGenus" || rank === "species" || rank === "subSpecies" ||
+          rank === "variety") &&
+        normalizeSpace(taxon.getAttribute("genus")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `genus '${
+            normalizeSpace(taxon.getAttribute("genus"))
+          }' contains invalid characters`,
+        );
+      }
+      if (
+        (rank === "subFamily" || rank === "tribe" || rank === "subTribe") &&
+        normalizeSpace(taxon.getAttribute("family")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `family '${
+            normalizeSpace(taxon.getAttribute("family"))
+          }' contains invalid characters`,
+        );
+      }
+      if (
+        rank === "subOrder" &&
+        normalizeSpace(taxon.getAttribute("order")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `order '${
+            normalizeSpace(taxon.getAttribute("order"))
+          }' contains invalid characters`,
+        );
+      }
+      if (
+        rank === "subClass" &&
+        normalizeSpace(taxon.getAttribute("class")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `class '${
+            normalizeSpace(taxon.getAttribute("class"))
+          }' contains invalid characters`,
+        );
+      }
+      if (
+        rank === "subPhylum" &&
+        normalizeSpace(taxon.getAttribute("phylum")).match(/[^a-zA-Z\-]/)
+      ) {
+        errors.push(
+          `phylum '${
+            normalizeSpace(taxon.getAttribute("phylum"))
+          }' contains invalid characters`,
+        );
+      }
+      if (!taxon.getAttribute("kingdom")) {
+        console.warn(
+          "Warning: treatment taxon is missing ancestor kingdom, defaulting to 'Animalia'",
+        );
+        output(
+          "# Warning: treatment taxon is missing ancestor kingdom, defaulting to 'Animalia'",
+        );
+      }
+    }
+    if (errors.length) {
+      throw new Error(
```

Comment:
RDF/XML?

Reply:
error message is copy-pasted from xslt

you are correct that that makes no sense anymore, I will change it